dangling pointer warning (`gcc-12`)

**Describe the bug**
[gcc-12 build](https://dev.azure.com/DARMA-tasking/DARMA/_build/results?buildId=32691&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=28db5144-7e5d-5c90-2820-8676d630d9d2) shows a new warning:
```
/vt/src/vt/group/region/group_list.cc: In member function 'virtual void vt::group::region::List::splitN(int, vt::group::region::Region::ApplyFnType) const':
/vt/src/vt/group/region/group_list.cc:164:31: warning: dangling pointer to an unnamed temporary may be used [-Wdangling-pointer=]
  164 |     for (int i = cur_idx; i < cur_max; i++) {
      |                               ^~~~~~~
/vt/src/vt/group/region/group_list.cc:162:22: note: unnamed temporary defined here
  162 |       size : std::min(size, cur_idx + child_size);
      |              ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Overall the code involved seems to be using references to `int` overzealously:
https://github.com/DARMA-tasking/vt/blob/be0a753bd3adbec6b02563d86f83597e35a8e7e9/src/vt/group/region/group_list.cc#L155-L162
**Expected behavior**
Remove the unnecessary `&`.

--- a/src/vt/group/region/group_list.cc
+++ b/src/vt/group/region/group_list.cc
@@ -97,7 +97,7 @@ List::List(List const& in_other, BoundType in_remove_extent) {
 }
 
 List::List(
-  BoundType const* const list, SizeType const& size, bool const& is_sorted
+  BoundType const* const list, SizeType const size, bool const is_sorted
 ) {
   ListType new_list;
   for (SizeType elm = 0; elm < size; elm++) {
@@ -128,7 +128,7 @@ List::List(
 }
 
 /*virtual*/ List::SplitRegionType List::split() const {
-  auto const& size = getSize();
+  auto const size = getSize();
   vtAssert(
     size >= 2, "Size must be at least 2 to split"
   );
@@ -153,12 +153,12 @@ List::List(
 }
 
 /*virtual*/ void List::splitN(int nsplits, ApplyFnType apply) const {
-  auto const& size = static_cast<int>(getSize());
-  auto const& num_splits = std::min(nsplits, size);
+  auto const size = static_cast<int>(getSize());
+  auto const num_splits = std::min(nsplits, size);
   int cur_idx = 0;
   for (auto split = 0; split < num_splits; split++) {
-    auto const& child_size = size / num_splits;
-    auto const& cur_max = split == num_splits - 1 ?
+    auto const child_size = size / num_splits;
+    auto const cur_max = split == num_splits - 1 ?
       size : std::min(size, cur_idx + child_size);
     ListType list;
     for (int i = cur_idx; i < cur_max; i++) {

--- a/src/vt/group/region/group_list.h
+++ b/src/vt/group/region/group_list.h
@@ -44,13 +44,15 @@
 #if !defined INCLUDED_VT_GROUP_REGION_GROUP_LIST_H
 #define INCLUDED_VT_GROUP_REGION_GROUP_LIST_H
 
+#include "vt/group/region/group_region.h"
+
 namespace vt { namespace group { namespace region {
 
 struct List : Region {
   explicit List(ListType const& in_list);
   explicit List(ListType&& in_list);
   List(List const& in_other, BoundType in_remove_extent);
-  List(BoundType const* const list, SizeType const& size, bool const& is_sorted);
+  List(BoundType const* const list, SizeType const size, bool const is_sorted);
 
   List(List const&) = default;
   List(List&&) = default;

